tests/GoogleAuthenticatorTest.php

```diff
@@ -33,52 +33,28 @@ public function testGenerateSecret()
         );
     }
 
-    public function testCheckCode()
-    {
-        $authenticator = new GoogleAuthenticator(6, 10, new \DateTime('2012-03-17 22:17:00'));
-        $this->assertTrue(
-            $authenticator->checkCode('3DHTQX4GCRKHGS55CJ', $authenticator->getCode('3DHTQX4GCRKHGS55CJ', strtotime('2012-03-17 22:17:00') / 30))
-        );
-    }
-
-    public function testCheckCodeTooOld()
-    {
-        $authenticator = new GoogleAuthenticator(6, 10, new \DateTime('2012-03-17 22:17:00'));
-        $this->assertFalse(
-            $authenticator->checkCode('3DHTQX4GCRKHGS55CJ', $authenticator->getCode('3DHTQX4GCRKHGS55CJ', strtotime('2012-03-17 22:18:00') / 30))
-        );
-    }
-
-    public function testCheckCodePlusOne()
-    {
-        $authenticator = new GoogleAuthenticator(6, 10, new \DateTime('2012-03-17 22:17:00'));
-        $this->assertTrue(
-            $authenticator->checkCode('3DHTQX4GCRKHGS55CJ', $authenticator->getCode('3DHTQX4GCRKHGS55CJ', strtotime('2012-03-17 22:17:30') / 30))
-        );
-    }
-
-    public function testCheckCodeMinusOne()
-    {
-        $authenticator = new GoogleAuthenticator(6, 10, new \DateTime('2012-03-17 22:17:00'));
-        $this->assertTrue(
-            $authenticator->checkCode('3DHTQX4GCRKHGS55CJ', $authenticator->getCode('3DHTQX4GCRKHGS55CJ', strtotime('2012-03-17 22:16:30') / 30))
-        );
-    }
-
-    public function testCheckCodeTooYoung()
+    /**
+     * @dataProvider testCheckCodeData
+     */
+    public function testCheckCode($expectation, $inputDate)
     {
         $authenticator = new GoogleAuthenticator(6, 10, new \DateTime('2012-03-17 22:17:00'));
-        $this->assertFalse(
-            $authenticator->checkCode('3DHTQX4GCRKHGS55CJ', $authenticator->getCode('3DHTQX4GCRKHGS55CJ', strtotime('2012-03-17 22:16:29') / 30))
+        $this->assertSame(
+            $expectation,
+            $authenticator->checkCode('3DHTQX4GCRKHGS55CJ', $authenticator->getCode('3DHTQX4GCRKHGS55CJ', strtotime($inputDate) / 30))
         );
     }
 
-    public function testCheckCodeWithInvalidDateArgument()
+    public static function testCheckCodeData(): array
     {
-        $authenticator = new GoogleAuthenticator(6, 10, new \DateTime());
-        $this->assertFalse(
-            $authenticator->checkCode('3DHTQX4GCRKHGS55CJ', $authenticator->getCode('3DHTQX4GCRKHGS55CJ', false))
-        );
+        return [
+            [false, '2012-03-17 22:16:29'],
+            [true, '2012-03-17 22:16:30'],
+            [true, '2012-03-17 22:17:00'],
+            [true, '2012-03-17 22:17:30'],
+            [false, '2012-03-17 22:18:00'],
+            [false, 'this date cannot be resolved and results into false'],
```

Comment:
I can add a generic description to the test (or data provider) itself, which I think is more useful